hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -402,19 +402,23 @@ public static Config loadFromFile(File configFile, Properties properties) throws
         checkTrue(properties != null, "properties can't be null");
 
         String path = configFile.getPath();
-        InputStream stream = new FileInputStream(configFile);
-        if (path.endsWith(".xml")) {
-            return applyEnvAndSystemVariableOverrides(
-                    new XmlConfigBuilder(stream).setProperties(properties).build().setConfigurationFile(configFile)
-            );
-        }
-        if (path.endsWith(".yaml") || path.endsWith(".yml")) {
-            return applyEnvAndSystemVariableOverrides(
-                    new YamlConfigBuilder(stream).setProperties(properties).build().setConfigurationFile(configFile)
-            );
-        }
+        try (InputStream stream = new FileInputStream(configFile)) {
+            final Config config;
+
+            if (path.endsWith(".xml")) {
+                config = new XmlConfigBuilder(stream).setProperties(properties).build();
+            } else if (path.endsWith(".yaml") || path.endsWith(".yml")) {
+                config = new YamlConfigBuilder(stream).setProperties(properties).build();
+            } else {
+                throw new IllegalArgumentException("Unknown configuration file extension");
+            }
 
-        throw new IllegalArgumentException("Unknown configuration file extension");
+            return applyEnvAndSystemVariableOverrides(config.setConfigurationFile(configFile));
+        } catch (FileNotFoundException e) {
+            throw e;
+        } catch (IOException e) {
+            throw ExceptionUtil.sneakyThrow(e);
+        }
```

Comment:
These catch blocks are not necessary, we already declare exceptions in the method signature

Reply:
> These catch blocks are not necessary, we already declare exceptions in the method signature

We declare the method can throw `FileNotFoundException`, but `close()` throws `IOException` which we don't catch. Therefore, ensure that `FileNotFoundException` is thrown normally, but any other `IOException` is thrown "sneakily".

Reply:
I'm still a fan of explicit exception declarations and handling (i.e. declare IOException here and handle in called locations) - since this method is only called from 2 places in production code, both of which already have some error handling in place, I think it's okay as you've presented it. Thanks for the info 👍